tasks/upload.py

```diff
@@ -959,25 +974,51 @@ def possibly_setup_webhooks(self, commit, repository_service):
                 else:
                     # service-level config value will be used instead in this case
                     webhook_secret = None
-                hook_result = async_to_sync(create_webhook_on_provider)(
-                    repository_service, webhook_secret=webhook_secret
-                )
-                hookid = hook_result["id"]
-                log.info(
-                    "Registered hook",
-                    extra=dict(
-                        repoid=commit.repoid, commit=commit.commitid, hookid=hookid
-                    ),
-                )
-                repository.hookid = hookid
+                if should_post_webhook:
+                    hook_result = async_to_sync(create_webhook_on_provider)(
+                        repository_service, webhook_secret=webhook_secret
+                    )
+                    hookid = hook_result["id"]
+                    log.info(
+                        "Registered hook",
+                        extra=dict(
+                            repoid=commit.repoid,
+                            commit=commit.commitid,
+                            hookid=hookid,
+                            action="SET",
+                        ),
+                    )
+                    repository.hookid = hookid
+                    repo_data["repo"]["hookid"] = hookid
+                else:
+                    async_to_sync(gitlab_webhook_update)(
+                        repository_service=repository_service,
+                        hookid=repository.hookid,
+                        secret=webhook_secret,
+                    )
+                    log.info(
+                        "Updated hook",
+                        extra=dict(
+                            repository_service=repository_service.service,
+                            repoid=repository.repoid,
+                            commit=commit.commitid,
+                            hookid=repository.hookid,
+                            action="EDIT",
+                        ),
+                    )
+
                 if webhook_secret is not None:
                     repository.webhook_secret = webhook_secret
-                repo_data["repo"]["hookid"] = hookid
-                return True
+
+                return True if should_post_webhook else False
```

Comment:
With the other suggestion above, I think it'd be a bit more readable to put return True in the `should_post_webhook` conditional, and false in the `else` in 993, wdyt?

Reply:
agree! updating